src/pages/api/breadcrumbs.ts

```diff
@@ -0,0 +1,74 @@
+import apiUrl from '../../utils/apiUrl';
+import { NextApiRequest, NextApiResponse } from 'next';
+
+type QueryObject = {[key: string]: string | string[]}
+type ReturnedData = {[key: string]: {[key: string]: string}}
+
+export default async (
+    req: NextApiRequest,
+    res: NextApiResponse,
+): Promise<void> => {
+    const returnedData = await fetchData(req.query);
+    const breadcrumbs = pathToCrumbs(req.query.path as string, returnedData);
```

Comment:
I feel like this should be possible to do in a single pass with just one iteration, which at least to me would make it more readable and perhaps even result in less code. Something like this (a little bit sloppy and untested):

```typescript
const pathFields = req.query.path.split('/');
const crumbs = [];

// This replaces the realPath, and is constructed gradually
// as we iterate over the fields of the path
let curPath = [];

for (const field of pathFields) {
  // Is this a variable field, e.g. [personId]?
  if (field.beginsWith('[') && field.endsWith(']')) {
    const fieldName = field.slice(1, -1);
    const fieldValue = req.query[fieldName];

    const label = await loadLabel(fieldName, fieldValue);

    // Add the value (i.e. the numeric ID) instead of the
    // name of this field to curPath. E.g.['organize', '[orgId]']
    // becomes ['organize', '1']
    curPath.push(fieldValue);

    crumbs.push({
      label: label,
      href: '/' + curPath.join('/'),
    });
  }
  else {
    curPath.push(field);
    crumbs.push({
      labelMsg: `misc.breadcrumbs.${field}`,
      href: '/' + curPath.join('/'),
    });
  }
}
```

The `loadLabel()` function is left out of the implementation above, but could be some sort of lookup table or something similar to what you're already doing, e.g.:

```typescript
async function loadLabel(fieldName, id) {
    if (fieldName == 'personId') {
      const person = await fetch(
            apiUrl(`/orgs/${orgId}/people/${query.personId}`),
        ).then((res) => res.json());

        // No need to return ID, as it was already passed to this
        // function and hence clearly known in the context that invoked it
        return `${person.data.first_name} ${person.data.last_name}`;
    }
    else if (fieldName == 'orgId') {
      // Do something similar
    }
    else {
      // Return something generic, e.g.:
      return `Object ${id}`;
    }
}
```